### ISSUE TITLE
Gitea API Problem

### ISSUE THREAD

**Issue by ykorzikowski** [decaporg/decap-cms#6997]
Hi, 

I am trying to set up decap with gitea. But I guess there is a bug with the usage of the repos/git/trees API. 

There is always a 404 for all content defined in netlify. In brwoser console, I can see, that a GET to `https://git.***.dev/api/v1/repos/***/blog-ng/git/trees/dev:elegantstack%2Fsite%2Fcontent%2Fassets` is fired. In this example I list my assets in the netlify app. 

When I do a curl to  `https://git.***.dev/api/v1/repos/***/blog-ng/git/trees/dev:elegantstack`, I get a response. But I cant add any more subdirectories. curl to  `https://git.***.dev/api/v1/repos/***/blog-ng/git/trees/dev:elegantstack%2Fsite` will result in a 404 again. 

The path for the assets is correct of course. Maybe the api usage need to be changed to query the root path wit recursive=1?

**Comment by derBlume** [comment]
Hey everyone! Any news on this? I have the exact same problem @ykorzikowski describes (I'm on Gitea 1.20.5). Maybe @denyskon has thoughts on this, as he authored the gitea plugin for Decap and is also involved with gitea itself.

**Comment by ki9us** [comment]
> Seems that there is also no file-path option for the api you use (or they removed it).

Agreed, this looks like a breaking change that was never fixed.  Gitea backend isn't working at all as a result.  

I always get a 404 at `https://{domain}/api/vi/repos/{owner}/{repo}/git/trees/{branch}:{path}`

But if I use a sha of a recent commit, I get a 200 from `https://{domain}/api/vi/repos/{owner}/{repo}/git/trees/{sha}`.  

I'm using `v1.22.1`.

**Comment by ki9us** [comment]
Oh, you're right.  So it seems to only 404 in subdirs.  

If I set sha to `main` or `main:_src` it works, but `main:_src/pages` does not even though `_src/pages` is a valid path in my repo.  I tried `curl`ing these with both a forward slash and its encoding (%2F) and both threw 404s.  

I am experimenting with my instance's /api/swagger so maybe it's a gitea issue.  

@denyskon Can you try to replicate this with a subdir?

**Comment by ki9us** [comment]
```sh
$ curl -X 'GET' \
  'https://gitea.com/api/v1/repos/gitea/gitea-mirror/git/trees/main%3Apublic%2Fassets' \
  -H 'accept: application/json'
{"sha":"3ba6530bf5a26ca2d93c9c6b2a8cf81c5163fe69","url":"https://gitea.com/api/v1/repos/gitea/gitea-mirror/git/trees/3ba6530bf5a26ca2d93c9c6b2a8cf81c5163fe69","tree":[{"path":"img","mode":"040000","type":"tree","size":0,"sha":"fc924277fa697dd7b636bcd747fc5660ae910014","url":"https://gitea.com/api/v1/repos/gitea/gitea-mirror/git/trees/fc924277fa697dd7b636bcd747fc5660ae910014"}],"truncated":false,"page":1,"total_count":1}
```
```sh
$ curl -X 'GET'   'https://mygitea.tld/api/v1/repos/myowner/myrepo/git/trees/main%3A_src%2Fpages'   -H 'accept: application/json'
404 page not found
```

Well it doesn't happen on the official instance... just mine.

**Comment by ki9us** [comment]
Fixed it for me!  I was adding a trailing slash in my reverse proxy, nginx: 

**Does not work:** 
```nginx
upstream gitea {
    server localhost:3000;
}
server {
    ...
    location / {
        ...
        proxy_pass http://gitea/;
    }
}
```

**Works:** 
```nginx
upstream gitea {
    server localhost:3000;
}
server {
    ...
    location / {
        ...
        proxy_pass http://gitea;
    }
}
```

Got a clue from go-gitea/gitea#6187

@ykorzikowski @derBlume Check your reverse proxy if you have one.

**Comment by ykorzikowski** [comment]
@ki9us I can confirm its working now :)

**Comment by martinjagodic** [comment]
@ykorzikowski does that mean we can close this issue?
